.github/workflows/test.yml

```diff
@@ -87,9 +105,9 @@ jobs:
   # This job reports the results of the test matrix above
   test:
     if: always()
-    needs: test-matrix
+    needs: [test-matrix, all-features, oldstable]
     runs-on: ubuntu-latest
     steps:
-      - if: needs.test-matrix.result != 'success'
+      - if: contains(needs.*.result, 'failure')
```

Comment:
Here are [the docs](https://docs.github.com/en/actions/learn-github-actions/contexts#needs-context) for anyone wondering what the possible states are.

There is no reason to fail the build for `skipped`. But maybe we should test the `cancelled` case?

Reply:
Yeah cancelled probably should fail. You don't want to allow people to cancel all the stops to bypass the merge checks.